### ISSUE TITLE
Wrong string missing method when defining with round brackets

### ISSUE THREAD

**Issue by ttoommxx** [pylint-dev/pylint#9494]
### Bug description

```python
In Python it is perfectly fine to define a string in the following way:

s = (
"hello"
)
```
but when doing that and then successively calling a string method on `s`, for example `s.upper()` Pylint will erroneously complain that I am calling a string method on a typle, giving me error E1101:no-member
```


### Configuration

_No response_

### Command used

```shell
Using Pylint extension within VS Code
```

### Pylint output

```python
error E1101:no-member
```

### Expected behavior

Pylint should not say anything because all operations are perfectly fine

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.12.2 (main, Feb 21 2024, 00:00:00) [GCC 13.2.1 20231205 (Red Hat 13.2.1-6)]
```

### OS / Environment

Fedora 39
